Add keyboard control for cursor movement and node rotation

neo-sagittal opened the pull request:
This pull request introduces keyboard controls to allow users to move a cursor across nodes and rotate nodes using arrow keys and shift + arrow keys, respectively. The changes include the implementation of a `KeyboardInput` class to handle these inputs and a `CursorIndicator` class to visually indicate the current node selected by the keyboard. This enhancement fixes #16 by enabling keyboard navigation and interaction with nodes, providing a more accessible and efficient user experience.

palako committed:
Add keyboard control for node navigation

Implement keyboard controls to allow users to navigate and rotate
nodes using arrow keys. The cursor starts at the first node and can
move in the direction of the arrow keys, skipping non-adjacent nodes.
If no node exists in the pressed direction, the cursor blinks to
indicate immobility. Holding Shift while pressing arrow keys rotates
the node in the direction pressed, adhering to legal move rules.
A subtle white line indicates the current node under keyboard control.

Fixes #16

palako commented on Assets/Scripts/View/Control/KeyboardInput.cs:
```diff
@@ -0,0 +1,212 @@
+using System.Linq;
+using Core.Data;
+using UnityEngine;
+using View.Game;
```
@neo-sagittal you need to include View.Items. Also, take a look at the definition of the Point class in Assets/Scripts/Core/Data/Point.cs, since to make sure you refer to things properly, including case-sensitive names.
Since you are at it, overload the == and != operators in that class, since it already has an Equals method

neo-sagittal replied:
I've pushed the changes as required to [this commit](https://github.com/palako/nodulus/commit/4b0f8f74151bc331b88bbb79e8b6bb03335b7d62).

palako committed:
Add operator overloads and fix case issues

Implemented operator overloads for '==' and '!=' in the Point class
within Assets/Scripts/Core/Data/Point.cs to enhance comparison
operations, complementing the existing Equals method. This change
ensures more intuitive equality checks between Point instances.

Updated KeyboardInput.cs to include View.Items and corrected
case-sensitive references to Point's properties, ensuring proper
functionality and adherence to naming conventions. These adjustments
address issues with cursor movement and node rotation logic.

palako commented on Assets/Scripts/View/Control/KeyboardInput.cs:
```diff
@@ -0,0 +1,213 @@
+using System.Collections.Generic;
```
@neo-sagittal I think instead of this class this should be in GameController. Or if this is behavioural, at least the capturing of the keyboard events should be done there and then from there use the logic in this file. It's in Assets/Scripts/View/Control/GameController.cs

neo-sagittal replied:
I've pushed the changes as required to [this commit](https://github.com/palako/nodulus/commit/aef7c22e6c916795e7227bd1b8bc3fe9c534ec6a).

palako committed:
Refactor keyboard input handling

This commit refactors the handling of keyboard input by moving the
keyboard event capturing logic from the KeyboardInput class to the
GameController class. This change aligns with the architectural
suggestion to centralize input handling within the GameController.

The GameController now initializes a KeyboardInput component and
captures keyboard events to control cursor movement and node rotation.
The logic for moving the cursor and rotating nodes has been extracted
into public methods within the KeyboardInput class, allowing the
GameController to invoke these actions based on user input.

This refactoring improves the separation of concerns by keeping the
input handling within the GameController while maintaining the
behavioral logic within the KeyboardInput class. It also enhances
code readability and maintainability by clearly delineating the
responsibilities of each class.
